### PR TITLE
Some pretty cli for trace

### PR DESCRIPTION
**Pull request by manasdk**
See the output to see what the client does.

```
$ st2 trace list
+--------------------------+----------------------------------------------------+-------------------------------+
| id                       | trace_tag                                          | start_timestamp               |
+--------------------------+----------------------------------------------------+-------------------------------+
| 55d3c9b132ed35435dd6a68f | trigger_instance-55d3c9b132ed35435dd6a68e          | Wed, 19 Aug 2015 00:11:29 UTC |
| 55d3c9ac32ed35435dd6a68d | trigger_instance-55d3c9ac32ed35435dd6a68c          | Wed, 19 Aug 2015 00:11:24 UTC |
| 55d3c9a732ed35435dd6a68b | trigger_instance-55d3c9a732ed35435dd6a68a          | Wed, 19 Aug 2015 00:11:19 UTC |
| 55d3c9a532ed35435dd6a689 | trigger_instance-55d3c9a532ed35435dd6a688          | Wed, 19 Aug 2015 00:11:17 UTC |
| 55d3c9a232ed35435dd6a687 | trigger_instance-55d3c9a232ed35435dd6a686          | Wed, 19 Aug 2015 00:11:14 UTC |
| 55d3c99f32ed35435dd6a685 | trigger_instance-55d3c99f32ed35435dd6a684          | Wed, 19 Aug 2015 00:11:11 UTC |
| 55d3c99f32ed35435dd6a683 | trigger_instance-55d3c99f32ed35435dd6a682          | Wed, 19 Aug 2015 00:11:11 UTC |
| 55d3d29732ed35435dd6a691 | TestPassiveSensor-ec4cf37b7bc94b4aab66cc033f1fbefa | Wed, 19 Aug 2015 00:49:27 UTC |
+--------------------------+----------------------------------------------------+-------------------------------+
```

```
$ st2 trace get 55d3d29732ed35435dd6a691
id: 55d3d29732ed35435dd6a691
trace_tag: TestPassiveSensor-ec4cf37b7bc94b4aab66cc033f1fbefa
start_timestamp: 2015-08-19T00:49:27.034587Z
+--------------------------+-----------------+-----------------------------+
| id                       | type            | updated_at                  |
+--------------------------+-----------------+-----------------------------+
| 55d3d29732ed35435dd6a690 | triggerinstance | 2015-08-19T00:49:27.034665Z |
| 55d3c99e32ed354372c0f7fe | rule            | 2015-08-19T00:49:27.057020Z |
| 55d3d29732ed35435dd6a693 | execution       | 2015-08-19T00:49:27.102228Z |
| 55d3d29732ed35435dd6a694 | triggerinstance | 2015-08-19T00:49:27.285334Z |
+--------------------------+-----------------+-----------------------------+
```

```
$ st2 trace list -c TestPassiveSensor-ec4cf37b7bc94b4aab66cc033f1fbefa
id: 55d3d29732ed35435dd6a691
trace_tag: TestPassiveSensor-ec4cf37b7bc94b4aab66cc033f1fbefa
start_timestamp: 2015-08-19T00:49:27.034587Z
+--------------------------+-----------------+-----------------------------+
| id                       | type            | updated_at                  |
+--------------------------+-----------------+-----------------------------+
| 55d3d29732ed35435dd6a690 | triggerinstance | 2015-08-19T00:49:27.034665Z |
| 55d3c99e32ed354372c0f7fe | rule            | 2015-08-19T00:49:27.057020Z |
| 55d3d29732ed35435dd6a693 | execution       | 2015-08-19T00:49:27.102228Z |
| 55d3d29732ed35435dd6a694 | triggerinstance | 2015-08-19T00:49:27.285334Z |
+--------------------------+-----------------+-----------------------------+
```

### CODE REVIEW

File: st2client/st2client/commands/trace.py [hunk 1]
```diff
@@ -37,7 +43,42 @@ def __init__(self, description, app, subparsers, parent_parser=None):
             })
 
 
-class TraceListCommand(resource.ResourceCommand):
+class SingleTraceDisplayMixin(object):
```
Comment (lakshmi-kannan) [inline st2client/st2client/commands/trace.py]: Ghetto.

**Comment by lakshmi-kannan** [comment]
Wondering if we should switch id and type columns :P.

**Comment by manasdk** [comment]
> Wondering if we should switch id and type columns :P.

Likely we can switch stuff around. Almost all our lists have `id` as the first field so for starters it is better to be consistent with that for now. Lets have some user feedback and then we can make the Ux changes.

File: st2client/st2client/commands/trace.py (hunk 1 above)
Comment (manasdk) [inline st2client/st2client/commands/trace.py]: Whatever works.

**Comment by Kami** [comment]
```bash
$ st2 trace list -c TestPassiveSensor-ec4cf37b7bc94b4aab66cc033f1fbefa
id: 55d3d29732ed35435dd6a691
trace_tag: TestPassiveSensor-ec4cf37b7bc94b4aab66cc033f1fbefa
start_timestamp: 2015-08-19T00:49:27.034587Z
+--------------------------+-----------------+-----------------------------+
| id                       | type            | updated_at                  |
+--------------------------+-----------------+-----------------------------+
| 55d3d29732ed35435dd6a690 | triggerinstance | 2015-08-19T00:49:27.034665Z |
```

I assume that in this output `id` refers to the traced resource.

If that's the case, imo, it would make it more clear if the `id` header field is called `resource_id` and we display it after the resource type. Otherwise it's a bit confusing (at least to me) and not immediately clear to what this id refers to.

**Comment by manasdk** [comment]
Every list of our has `id` as column 1. Pretty much everywhere `id == resource_id` so I picked consistency. 

Also we make some assumptions like https://github.com/StackStorm/st2/blob/master/st2client/st2client/formatters/table.py#L49 which makes it preferable to use `id` as the 1st field and to retain the name. I don't necessarily agree with those impl choices but am very wary of changing the formatter code. It would be nice to be able to A/B test the choice and see which one works better. 

At this point I will wait for a few more folks to chime-in and see which direction consensus goes and will change if we all mostly agree on swapping `id` and `type`

**Comment by Kami** [comment]
@manasdk The difference with other operations (list actions, list rules, list executions, etc.) is that in that case `id` refers to the resource on which you are operating (action / rule / executions).

In the trace case though, you list traces so you would assume the `id` refers to the trace, but it actually refers to the resource trace is point to so this is what makes it confusing and not consistent (as mentioned above, please correct me if I'm wrong and if that id actually refers to the trace and not the resource - if it does, then we are OK :)).

**Comment by manasdk** [comment]
> The difference with other operations (list actions, list rules, list executions, etc.) is that in that case id refers to the resource on which you are operating (action / rule / executions).

Not really. The get for a workflow execution (which this is modeled on) has similar behavior. 

```
$ st2 execution get 55d3fd82fa4bf55edf94934b
id: 55d3fd82fa4bf55edf94934b
action.ref: st2cd.st2_pkg_f20
status: succeeded
start_timestamp: 2015-08-19T03:52:32.549801Z
end_timestamp: 2015-08-19T03:55:13.898937Z
+--------------------------+-----------+-------------------+-------------------------+-------------------------------+
| id                       | status    | task              | action                  | start_timestamp               |
+--------------------------+-----------+-------------------+-------------------------+-------------------------------+
| 55d3fd83fa4bf55ed053c8dd | succeeded | get_build_server  | linux.dig               | Wed, 19 Aug 2015 03:52:35 UTC |
| 55d3fd85fa4bf55ed053c8df | succeeded | clone_repo        | st2cd.git_clone         | Wed, 19 Aug 2015 03:52:37 UTC |
| 55d3fd8dfa4bf55ed053c8e1 | succeeded | version           | st2cd.st2_version       | Wed, 19 Aug 2015 03:52:45 UTC |
| 55d3fd8efa4bf55ed053c8e3 | succeeded | echo_version      | core.local              | Wed, 19 Aug 2015 03:52:46 UTC |
| 55d3fd90fa4bf55ed053c8e5 | succeeded | version_hack      | st2cd.version_hack      | Wed, 19 Aug 2015 03:52:48 UTC |
| 55d3fd91fa4bf55ed053c8e8 | succeeded | make_rpms         | st2cd.make_rpms         | Wed, 19 Aug 2015 03:52:49 UTC |
| 55d3fdf3fa4bf55ed053c8ea | succeeded | check_pkgs        | st2cd.check_pkgs        | Wed, 19 Aug 2015 03:54:27 UTC |
| 55d3fdf4fa4bf55ed053c8ec | succeeded | create_build_dir  | core.remote             | Wed, 19 Aug 2015 03:54:28 UTC |
| 55d3fdf5fa4bf55ed053c8ee | succeeded | update_st2_deploy | st2cd.update_st2_deploy | Wed, 19 Aug 2015 03:54:29 UTC |
| 55d3fdf7fa4bf55ed053c8f0 | succeeded | rsync_st2_deploy  | linux.rsync             | Wed, 19 Aug 2015 03:54:31 UTC |
| 55d3fdfafa4bf55ed053c8f3 | succeeded | rsync_rpms        | linux.rsync             | Wed, 19 Aug 2015 03:54:34 UTC |
| 55d3fdfbfa4bf55ed053c8f5 | succeeded | cp_st2_deploy     | core.remote_sudo        | Wed, 19 Aug 2015 03:54:35 UTC |
| 55d3fdfdfa4bf55ed053c8f7 | succeeded | cp_rpms           | core.remote_sudo        | Wed, 19 Aug 2015 03:54:37 UTC |
| 55d3fdfefa4bf55ed053c8f9 | succeeded | cp_rpms_repo      | core.remote_sudo        | Wed, 19 Aug 2015 03:54:38 UTC |
| 55d3fe00fa4bf55ed053c8fb | succeeded | rebuild_repo      | core.remote_sudo        | Wed, 19 Aug 2015 03:54:39 UTC |
| 55d3fe1ffa4bf55ed053c8fe | succeeded | update_current    | core.remote_sudo        | Wed, 19 Aug 2015 03:55:11 UTC |
| 55d3fe20fa4bf55ed053c900 | succeeded | clean_repo        | st2cd.git_clean         | Wed, 19 Aug 2015 03:55:12 UTC |
+--------------------------+-----------+-------------------+-------------------------+-------------------------------+
```

But believe me I get your point. I hear it loud and clear and almost agree with you and FWIW was one of the options I considered. This table will guaranteed undergo a redesign in the next week or so. It is not final but is functional at this point so until I gather some more feedback on this and other aspects of the table I am reluctant to make any changes.

> but it actually refers to the resource trace is point to so this is what makes it confusing and not consistent

No you are right it is id of a TraceComponent so a resource/entity in StackStorm which is not a trace. If I explicitly tell you that the id is the id of the component then are you still confused? In fact what else could it be?

**Pull request merged**